shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -475,6 +474,17 @@ final class TsTypegenCodeBuilder {
             SourceCode.equalSign ++ bodyType)
           uvAppliedName
         }
+
+        // if a tuple is a recursive type then it is aliased
+        // so we must use the alias name for the argument type
+        if (funcArg) {
+          // a function argument is always a tuple hence we
+          // should spread it across the argument variable
+          val arg = typegenCtx.termScope.declareRuntimeSymbol("arg")
+          (SourceCode(s"...$arg: ") ++ recTypeCode).parenthesized
+        } else {
+          recTypeCode
+        }
```

Comment:
The correct logic is to special-case _function types_ to display the argument tuple inline. Your special-casing recursive types does not make much sense. For example, the tuple type could be hiding in an applied type or in a type variable bound. You should make sure that the behavior is always correct, no add special cases to fit the current test you're looking at.